docs/source/code-generation/codegen-configuration.mdx

```diff
@@ -0,0 +1,359 @@
+
+| Value | escription |
+| ----- | ----------- |
+| `clientControlledNullability` | If enabled, codegen will understand and parse Client Controlled Nullability. Read the [RFC](https://github.com/graphql/graphql-spec/issues/867) for more detail. |
+| `legacySafelistingCompatibleOperations` | If enabled, the generated operations will be transformed using a method that attempts to maintain compatibility with the legacy behavior from [`apollo-tooling`](https://github.dev/apollographql/apollo-tooling) for registering persisted operation to a safelist. *Note: Safelisting queries is a deprecated feature of Apollo Server that has reduced support for legacy use cases. This option may not work as intended in all situations.* |
+
+<ExpansionPanel title="Sample JSON">
+
+```json
+"experimentalFeatures": {
+	"clientControlledNullability": false,
+	"legacySafelistingCompatibleOperations": false
+}
+```
+
+</ExpansionPanel>
+
+<ExpansionPanel title="Sample Swift">
+
+```swift
+let configuration = ApolloCodegenConfiguration(
+	// Other properties not shown
+	experimentalFeatures: ApolloCodegenConfiguration.ExperimentalFeatures(
+		clientControlledNullability: false,
+		legacySafelistingCompatibleOperations: false)
+)
+```
+
+</ExpansionPanel>
+
+### Schema download configuration
+
+An optional step in the code generation process is to fetch a GraphQL schema from a remote server. This step ensures that you always have an up-to-date schema on which to base your operations and it eliminates the need to manually download the schema outside of the automated process.
+
+The properties to configure the schema download are:
+
+| Property Name | Description |
+| ------------- | ----------- |
+| `downloadMethod` | How to download the schema. |
+| `downloadTimeout` | The maximum time (in seconds) to wait before indicating that the download timed out. |
+| `headers` | Any additional headers to include when retrieving your schema. This is useful when the schema download requires authentication. |
+| `outputPath` | The local path where the downloaded schema should be written to. |
+
+<ExpansionPanel title="Sample JSON">
+
+```json
+"schemaDownloadConfiguration": {
+	"downloadMethod": {
+		"introspection": {
+			"endpointURL": "https://server.com",
+			"httpMethod": {
+				"POST": {}
+			},
+			"includeDeprecatedInputValues": false,
+			"outputFormat": "SDL"
+		}
+	},
+	"downloadTimeout": 60,
+	"headers": [],
+	"outputPath": "./graphql/"
+}
+```
+
+</ExpansionPanel>
+
+<ExpansionPanel title="Sample Swift">
+
+```swift
+let configuration = ApolloCodegenConfiguration(
+	// Other properties not shown
+	schemaDownloadConfiguration: ApolloSchemaDownloadConfiguration(
+		using: .introspection(
+			endpointURL: URL(string: "https://server.com")!),
+		timeout: 60.0,
+		headers: [],
+		outputPath: "./graphql/")
+)
+```
+
+</ExpansionPanel>
+
+#### Download method
+
+There are two supported sources for fetching a GraphQL schema:
+
+##### 1. Apollo server registry
```

Comment:
I cannot find a place to link to where there is concise information about the registry. If anyone knows where to find that please let me know.